### DISCUSSION TITLE
How can I convert Chen2020 to 18650 NMC/graphtie cell

### DISCUSSION THREAD

**Discussion by gutsfromberserk1** [pybamm-team/PyBaMM#3366] [Q&A]
Hi,
I was trying to convert Chen2020 parameter set to 18650 battery by changing 
params["Electrode height [m]"]=0.065
params["Electrode width [m]"]=0.018
params['Cell volume [m3]']=1.6*10**-5
 and measure the voltage output with different temperatures and frequencies:
```ruby
import pybamm 
import numpy as np
import matplotlib.pyplot as plt
pointsperperiod=1024
period=10
amplitude=10**-3
temperature=-25

model=pybamm.lithium_ion.DFN()
chemistry=pybamm.parameter_sets.Chen2020
params = pybamm.ParameterValues(chemistry)
params["Electrode height [m]"]=0.065
params["Electrode width [m]"]=0.018
params['Cell volume [m3]']=1.6*10**-5
params['Ambient temperature [K]']=temperature+273.15

freq_array=np.logspace(-5,3,20)

for freq in freq_array:
    current=lambda t: amplitude * pybamm.sin(2 * np.pi * freq * t)
    params['Current function [A]']=current
    time_array= np.linspace(0, period/freq, pointsperperiod*period)
    solver=pybamm.CasadiSolver()
    sim = pybamm.Simulation(model, parameter_values=params,solver=solver)
    sim.solve(t_eval=time_array)
    solution = sim.solution
    V = solution["Terminal voltage [V]"]
    C = solution["Current [A]"]
    array_v=np.array(V.data)
    array_c=np.array(C.data)
    neg_c=-1*array_c
    plt.subplot(121)
    plt.plot(time_array[:len(neg_c)],array_v)
    plt.title("Voltage")
    plt.xlabel("Time (s)")
    plt.subplot(122)
    plt.plot(time_array[:len(neg_c)],neg_c)
    plt.title("Current")
    plt.xlabel("Time (s)")
    plt.show() 
```

When you run this, after the first frequency, both current and voltage gets distorted. You can reduce the distorted freq number by reducing the amplitude and for temperatures higher than -5C, amplitude=10**-3 works perfectly but below it the results get distorted.
For capacity values, I think I have checked and they are almost same with a normal 18650 cell. Do anyone know how to fix this? Or my cell volume and/or electrode properties are wrong?

Thanks,

**Comment by brosaplanella** [comment]
Note that the only parameter that depends on temperature in Chen 2020 is the reaction rate, which was only measured at higher temperatures (above room temperature) so at -25degC it is quite an extrapolation and therefore parameters are not reliable. ORegan2022 might work better but, again, parameters were only measured down to 0degC so you might face similar problems.

Note as well that you are not using a thermal model here, so your battery does not heat up and the cell volume does not play any role.

**Reply by gutsfromberserk1** [reply]
Thanks for the reply. What do you suggest for this kind of temperature dependent measurements? And also, I did not know there is a thermal model. So, could you explain what is it?

**Reply by brosaplanella** [reply]
If you are trying to capture the temperature effects, then you should certainly use a parameter set that has been calibrated to work at such temperatures. Unfortunately I am not aware of any parameter sets going to such low temperatures (in our work the minimum has been 0 degC).

[This notebook](https://docs.pybamm.org/en/latest/source/examples/notebooks/models/thermal-models.html), explains the various thermal models in PyBaMM.
